pkg/kv/kvserver/replica_raft.go

```diff
@@ -200,6 +202,12 @@ func (r *Replica) evalAndPropose(
 		// Fork the proposal's context span so that the proposal's context
 		// can outlive the original proposer's context.
 		proposal.ctx, proposal.sp = tracing.ForkSpan(ctx, "async consensus")
+		if proposal.sp != nil {
+			// We can't leak this span if we fail to hand the proposal to the
+			// replication layer, so finish it later in this method if we are to
+			// return with an error. (On success, we'll reset this to a noop).
+			maybeFinishSpan = proposal.sp.Finish
```

Comment:
Would this do what you want it to do?

https://go.dev/play/p/c_1PTKWUmBx

Reply:
😶‍🌫️ thanks for catching that, fixed. Annoying that making this testable is a large refactor away.